arch/xtensa/src/common/xtensa_exit.c

```diff
@@ -145,7 +145,7 @@ void up_exit(int status)
 
   /* Then switch contexts */
 
-  xtensa_context_restore(tcb->xcp.regs);
+  xtensa_context_restore(&tcb->xcp.regs);
```

Comment:
why need change here?

Reply:
We need to pass the address of the saving area because we need it a bit below to get the coprocessor saving area.
Otherwise we will be writing in an incorrect position and corrupt the stack. For exapmle this will make thr system crash when a task exits.

Reply:
Ok, should we move the cp state to stack like other arch too?

Reply:
I will review all the CP state handling after this PR, there are other places that I think need some cleanup.

Reply:
how about we merge this PR until you finish cp state cleanup?

Reply:
Yes, please.  I plan to do that after this PR.